src/models/Util.ts

```diff
@@ -78,6 +80,28 @@ export const Util = {
 
     isTechnicalCivilisation(civilisation: Civilisation): boolean {
         return civilisation.gameVersion === GameVersion.TECHNICAL;
-    }
+    },
+
+    isRandomCiv(civilisation: Civilisation): boolean {
+        return civilisation.name.toUpperCase() === "RANDOM";
+    },
 
+    setRandomCivilisation(playerEvent: PlayerEvent, draftId: string, draftStore: DraftsStore) {
+        const maxCivilisationIndex = Civilisation.ALL.length - 1;
+        const randomCivIndex = Math.floor(Math.random() * maxCivilisationIndex);
```

Comment:
Math.random() yields [0,1),  so `Math.floor(Math.random() * array.length)` yields a valid array index – the `- 1` leads to the last civ never being picked